os/arch/arm/src/amebasmart/amebasmart_serial.c

```diff
@@ -1312,40 +1296,25 @@ static void amebasmart_serial_pmnotify(FAR struct pm_callback_s *cb, int domain,
 #ifdef CONFIG_PM
 static int amebasmart_serial_pmprepare(FAR struct pm_callback_s *cb, int domain, enum pm_state_e pmstate)
 {
-	switch (pmstate)
-	{
-		case PM_NORMAL:
-		case PM_IDLE:
-		case PM_STANDBY:
-			break;
-
-		case PM_SLEEP:
-			/* UART0_DEV: hp uart0
-			 * UART1_DEV: hp uart1
-			 * UART2_DEV: hp uart2
-			 * UART3_DEV: hp uart3_bt
-			 * LOGUART_DEV: KM0 log uart */
+	if (pmstate == PM_SLEEP) {
 #ifdef CONFIG_RTL8730E_UART0
-			if ((g_uart0priv.txint_enable) && (!serial_writable(sdrv[0]))) {		/* If Tx init enable and FIFO not empty */
-					return ERROR;
-			}
+		if ((g_uart0priv.txint_enable) && (!serial_writable(sdrv[0]))) {		/* If Tx init enable and FIFO not empty */
```

Comment:
amebasmart_serial_pmprepare
amebasmart_spi_pmprepare

Please make changes so that serial and spi do not use the pm_register function, just like i2c and i2s.

You can raise new PR about it.

Reply:
@ewoodev , but these checks are from driver as a last protection, to prevent from system going to sleep, if some app is still running with utilizing those drivers.
I2C and I2S are for pre/post sleep process, both are different.